documentation/images/plantuml/boardfarm_prplMesh_classes.puml

```diff
@@ -0,0 +1,165 @@
+@startuml
+class LinuxDevice {
+    string tftp_dir
+
+    check_status()
+    get_interface_ipaddr()
+    get_interface_ip6addr()
+    get_interface_macaddr()
+    get_seconds_uptime()
+    enable_ipv6()
+    set_static_ip()
+    disable_ipv6()
+    release_dhcp()
+    check_access_url()
+    set_printk()
+    prefer_ipv4()
+    ping()
+    traceroute()
+    is_link_up()
+    set_link_state()
+    add_new_user()
+    copy_file_to_server()
+    gzip_str()
+    ip_neigh_flush()
+    sudo_sendline()
+    set_cli_size()
+    wait_for_linux()
+    get_dns_server_upstream()
+    get_nf_conntrack_conn_count()
+    get_proc_vmstat()
+    wait_for_network()
+    get_memfree()
+    start_tinyproxy()
+    take_lock()
+    release_lock()
+    perform_curl()
+}
+
+class OpenWrtRouter {
+    string promt
+    string uprompt
+    string uboot_eth
+    bool linux_booted
+    bool saveenv_safe
+    string lan_gmac_iface
+    string lan_iface
+    string wan_iface
+    bool tftp_server_int
+    bool flash_meta_booted
+    bool has_cmts
+    string cdrouter_config
+    uint uboot_net_delay
+    bool routing
+    IPv4Network lan_network
+    IPv4Address lan_gateway
+    string tmpdir
+
+    __init__()
+    get_file()
+    tftp_get_file()
+    tftp_get_file_uboot()
+    prepare_file()
+    install_package()
+    wait_for_boot()
+    network_restart()
+    firewall_restart()
+    get_wan_iface()
+    get_wan_proto()
+    setup_uboot_network()
+    config_wan_proto()
+    enable_mgmt_gui()
+    enable_ssh()
+    uci_allow_wan_http()
+    uci_allow_wan_ssh()
+    uci_allow_wan_https()
+    uci_forward_traffic_redirect()
+    uci_forward_traffic_rule()
+    wait_for_mounts()
+    get_dns_server()
+    get_user_id()
+    collect_stats()
+    parse_stats()
+}
+
+class rax40 {
+    string model
+    string wan_iface
+    string lan_iface
+    IPv4Network lan_network
+    IPv4Address lan_gateway
+    IPv4Network wan_network
+    IPv4Address wan_gateway
+    prplMesh_agent [..]
+
+    kill_console_at_exit()
+    check_logs()
+    check_status()
+}
+
+class backhaul_sniffer {
```

Comment:
Although the terms fronthaul_sniffer and backhaul_sniffer are technically correct, I find them confusing. I would prefer to call them wired_sniffer and wireless_sniffer.

Reply:
Actually, difference between wireless and wired sniffer is in "listener" set up: 
- For wired we just have to call `tcpdump` and point ports + iface to listen to (probably setting proto will also make sense), just to minimize garbage amount and ease search.
- For wireless we have to additionally check monitor capability, put WLAN iface to monitor mode.

Most part of methods, like some abstract `wait_for_capability_report` will be same for both sniffers. So, we even may split it in 3 classes: 
- generic sniffer (has methods to work with `tcpdump`, which are common)
- wired sniffer, extends "generic sniffer", extends with constructor to bring wired `tcpdump`
- wireless sniffer, extends "generic sniffer" with wireless monitor set up methods + constructor

Reply:
Updated UML to demonstrate proposal of split up.